### PR TITLE
[WIP] Feat grpc

### PR DESCRIPTION
**Pull request by xujianhai666**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by wongoo** [comment]
how about create a new branch `grpc` and merge all pr about grpc to this branch?  @AlexStocks

File: go.mod [hunk 1]
```diff
@@ -49,6 +49,9 @@ require (
 	go.etcd.io/etcd v3.3.13+incompatible
 	go.uber.org/atomic v1.4.0
 	go.uber.org/zap v1.10.0
+	golang.org/x/tools v0.0.0-20190524140312-2c0ae7006135
```
Comment (zouyx) [inline go.mod]: use version+incompatible to specify version

example:
```
go.etcd.io/etcd v3.3.13+incompatible
```

File: protocol/grpc/client.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package grpc
+
+import (
+	"reflect"
+
+	"google.golang.org/grpc"
```
Comment (zouyx) [inline protocol/grpc/client.go]: i think boss will tell you to split this package :)

File: protocol/grpc/grpc_protocol.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+var grpcProtocol *GrpcProtocol
+
+type GrpcProtocol struct {
+	protocol.BaseProtocol
+	serverMap  map[string]*Server
+	serverLock sync.Mutex
+}
+
+func NewGRPCProtocol() *GrpcProtocol {
+	return nil
```
Comment (zouyx) [inline protocol/grpc/grpc_protocol.go]: why nil..?

**Comment by AlexStocks** [comment]
> how about create a new branch `grpc` and merge all pr about grpc to this branch? @AlexStocks

I do not know.

**Pull request closed without merging**